src/guidellm/preprocess/dataset.py

```diff
@@ -0,0 +1,224 @@
+import os
+from collections.abc import Iterator
+from enum import Enum
+from pathlib import Path
+from typing import Any, Callable, Optional, Union
+
+from datasets import Dataset
+from loguru import logger
+from transformers import PreTrainedTokenizerBase
+
+from guidellm.dataset import load_dataset as guidellm_load_dataset
+from guidellm.utils import IntegerRangeSampler, check_load_processor
+
+SUPPORTED_TYPES = {
+    ".json",
+    ".csv",
+    ".parquet",
+}
+
+
+class ShortPromptStrategy(str, Enum):
+    IGNORE = "ignore"
+    CONCATENATE = "concatenate"
+    PAD = "pad"
+
+
+def handle_ignore_strategy(
+    current_prompt: str,
+    min_prompt_tokens: int,
+    tokenizer: PreTrainedTokenizerBase,
+    **_kwargs,
+) -> Optional[str]:
+    if len(tokenizer.encode(current_prompt)) < min_prompt_tokens:
+        logger.warning("Prompt too short, ignoring")
+        return None
+    return current_prompt
+
+
+def handle_concatenate_strategy(
```

Comment:
The logic here might return tokens that are above the max the user set with the configs. Can we generalize the statistics and pass those into the handlers so the handlers can decide how best to fit the prompt to the desired range?